client/src/main/java/org/apache/rocketmq/client/producer/MQProducer.java

```diff
@@ -53,11 +51,9 @@ SendResult send(final Message msg, final MessageQueue mq) throws MQClientExcepti
     SendResult send(final Message msg, final MessageQueue mq, final long timeout)
         throws MQClientException, RemotingException, MQBrokerException, InterruptedException;
 
-    void send(final Message msg, final MessageQueue mq, final SendCallback sendCallback)
-        throws MQClientException, RemotingException, InterruptedException;
+    void send(final Message msg, final MessageQueue mq, final SendCallback sendCallback);
```

Comment:
There is an upgrade issue if the code has catched the thrown checked-exception, like  `MQClientException`.

We should treat this issue carefully ~